.github/workflows/ci.yml

```diff
@@ -91,6 +91,52 @@ jobs:
     - name: Publish to codecov
       uses: codecov/codecov-action@v3
 
+  run-visual-tests:
+    name: Visual Tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/cache@v2
+      id: npm_cache
+      with:
+        path: '**/node_modules'
+        key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
+    - name: Install NPM viewer dependencies
+      if: steps.npm_cache.outputs.cache-hit != 'true'
+      working-directory: viewer
+      run: yarn install --immutable
+
+    - name: Execute visual tests
+      id: viewerVisualTests
+      working-directory: viewer
+      run: 'yarn test:visual'
+      continue-on-error: true
+
+    - name: Check for failed visual test diffs
+      if: (always() && steps.viewerVisualTests.outcome == 'failure')
```

Comment:
why `always() && ...` ?

Reply:
No clue... Blindly ported from CI step from /examples, will look into it.